### PR TITLE
[ws-manager] Properly report FallbackToLogsOnError termination messages

### PR DESCRIPTION
**Pull request by csweichel**
This PR makes ws-manager interpret the log-based failure messages reported with `TerminationMessageFallbackToLogsOnError`. This leads to improved error messages when supervisor fails to start.

It does that by attempting to parse the failure message as newline-delimited JSON. Starting from the end of the failure message/log output, the first line that has a `message` field gets reported. If that line contains an `error` field, that field is reported, too. If parsing failed, or no lines of this kind were found, the whole failure message is returned.

### How to test
See unit test

### CODE REVIEW

File: components/ws-manager/pkg/manager/status.go [hunk 1]
```diff
@@ -605,7 +607,7 @@ func extractFailure(wso workspaceObjects) (string, *api.WorkspacePhase) {
 	}
 
 	status := pod.Status
-	if status.Phase == corev1.PodFailed {
+	if status.Phase == corev1.PodFailed && (status.Reason != "" || status.Message != "") {
```
Comment (AlexTugarev) [inline components/ws-manager/pkg/manager/status.go:610]: is it guaranteed that there is non-empty `status` in case a pod failed?

**Review by AlexTugarev — APPROVED** [review]
Sounds reasonable to me

**Review by akosyakov — APPROVED** [review]
LGTM

File: components/ws-manager/pkg/manager/status.go (hunk 1 above)
Comment (csweichel) [reply components/ws-manager/pkg/manager/status.go:610]: `pod.Status` is a struct (not a pointer) to a struct, hence cannot be `nil`, if that's what you mean.
If the `status` is empty (in the sense of Go's default value), `status.Phase == ""` and we won't enter here.

Most of those conditions - in fact the whole status machinery - is derived purely from observing system behaviour. Until recently we hadn't seen pod failure with an empty `reason` and/or `message`. Now that we do, we add a condition and test :)

Comment (AlexTugarev) [reply components/ws-manager/pkg/manager/status.go:610]: > Until recently we hadn't seen pod failure with an empty reason and/or message. Now that we do, we add a condition and test :)

that's great!

**Pull request merged**